Fix XCCDF support

Honny1 opened the pull request:
This PR fixes crashes when using the XCCDF report.

How to reproduce the problem: Use oscpa-report with XCCDF results.  
For example: `oscap-report < tests/test_data/xccdf-report.xml`

The crashes were caused by a missing part of the if condition. To avoid this problem, I created integration tests with all test files. And I discovered problems: 
* The first is that it is expected that the rule must be present in some of the groups.
* Missing selected rule definition, which should be a content issue that is fixed in later versions.

jan-cerny commented:
How can we reproduce the crash?

How will you prevent the bug from happening in future again?

jan-cerny reviewed (commented):
I confirm it fixes the traceback for me.

However, I'm slightly concerned about the 2 warnings I get:

```
[jcerny@thinkpad oscap-report{pr/134}]$ python3 -m openscap_report.cli results.xml > report.html
WARNING: This file is not valid ARF report!
WARNING: OVAL results "all ovals" not found!
```

Regarding the first warning, I think that it's expected that the XCCDF results file isn't a valid ARF report. Therefore, the warning shouldn't be there. I don't know the details about how the warning is generated and what is checked. But if some validation is performed for ARFs, you probably should perform a similar validation for XCCDF results. WDYT?

Regarding the second warning, I think it could be worded better, eg. "The given input doesn't contain OVAL results, OVAL details won't be shown in the report."

jan-cerny commented on openscap_report/scap_results_parser/oval_and_cpe_tree_builder.py:
```diff
@@ -105,7 +105,7 @@ def insert_oval_and_cpe_trees_to_rules(self, rules):
             if rule.oval_definition_id in self.oval_definitions:
                 rule.oval_definition = self.oval_definitions[rule.oval_definition_id]
-            rule_group = self.group_parser.rule_to_grup_id[rule.rule_id]
+            rule_group = self.group_parser.rule_to_grup_id.get(rule.rule_id, "")
```
Nitpick: there is a typo in the variable name. grup vs group

Honny1 commented:
Changes:
* Fixes variable typo,
* Adds XCCDF validation,
* Improves warning messages,
* New test to verify if a given input file is a valid ARF or XCCDF message.

jan-cerny commented on tests/test_data/plant_catalog.xml:
```diff
@@ -0,0 +1,291 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
I think that you can use much smaller test file to test the validator, I think a file that has just a few lines would be enough to test it.

jan-cerny commented on tests/unit_tests/test_data_structure.py:
```diff
@@ -9,7 +9,11 @@
+                         PATH_TO_SIMPLE_RULE_FAIL_ARF,
+                         PATH_TO_SIMPLE_RULE_FAIL_XCCDF,
+                         PATH_TO_SIMPLE_RULE_PASS_ARF,
+                         PATH_TO_SIMPLE_RULE_PASS_XCCDF, PATH_TO_XCCDF)
```
I think that having each item on a new line would be a better code style

Honny1 replied:
I am using to format and sort of imports tool name [isort](https://pycqa.github.io/isort/index.html). This is the output of `isort`. Isort is used in the test suite to avoid unused imports or not formatted imports.  I think formatting with `isort` is good.

jan-cerny replied:
aha, clever! thanks for explanation

If it's a generated code then it doesn't make sense to change it

jan-cerny commented on openscap_report/scap_results_parser/scap_results_parser.py:
```diff
@@ -6,22 +6,47 @@
 
 SCHEMAS_DIR = Path(__file__).parent / "schemas"
+ARF_SCHEMAS_PATH = 'arf/1.1/asset-reporting-format_1.1.0.xsd'
+XCCDF_1_1_SCHEMAS_PATH = 'xccdf/1.1/xccdf-schema.xsd'
```
XCCDF 1.1 format is extremely outdated, I wouldn't add support for it

jan-cerny commented on openscap_report/scap_results_parser/parsers/oval_result_parser.py:
```diff
@@ -30,7 +30,7 @@ def _get_oval_reports(self):
         reports = self.root.find('.//arf:reports', NAMESPACES)
         if reports is None:
-            raise MissingOVALResult
+            raise MissingOVALResult("all_OVALs")
```
why "all_OVALs"?

Honny1 replied:
Because there should be an ID of the OVAL result that is missing. But in this case, there are missing all OVAL results.

jan-cerny commented on openscap_report/scap_results_parser/exceptions.py:
```diff
@@ -11,3 +11,7 @@ class MissingProcessableRules(Exception):
+
+
+class NotSupportedReportingFormat(Exception):
+    """Raise when the given input isn't valid ARF report or XCCDF report"""
```
```suggestion
    """Raise when the given input isn't a valid ARF report or XCCDF report"""
```

jan-cerny reviewed (approved):
I have generated reports from both ARF and XCCDF 1.2 results and I have reviewed them. Also I don't experience trace back.

Honny1 commented:
Changes: 
* Rebased on the main branch
* Changes before the last rebase:
  * Reduced content of XML file `plant_catalog.xml`,
  * Fixed typos in expectation docstring and warnings texts,
  * Removed support for XCCDF 1.1. 


Nothing has changed since @jan-cerny's last comment.

jan-cerny commented:
I have verified that it still works OK after the rebase.
